index.js

```diff
@@ -1,9 +1,10 @@
 'use strict';
 module.exports = function (str) {
+	function escapeIfNotPrecededByEscapeChar(escapeChar, match, offset, string) { return string[offset - 1] === escapeChar ? match : escapeChar + match;}
 	return str.replace(/(?:\\*)?"([^"\\]*\\.)*[^"]*"/g, function (match) {
 		return match
 			.replace(/\\"/g, '"')            // unescape double-quotes
-			.replace(/([^\\])'/g, '$1\\\'')  // escape single-quotes
+			.replace(/'/g, escapeIfNotPrecededByEscapeChar.bind(null, '\\'))
```

Comment:
if javascript supported regex lookbehind it would have been really easy... but only lookahead is supported.
I found a solution which was very close:
replace line 6 with the following:
    .replace(/\\(?!([^\']))/g, '')  // remove all escape chars (\) followed by a single-quote (') 
    .replace(/\'/g, '\\\'');           // replace all single-quotes (') with escape+single-quote (\')
but... this solution has a bug for strings that end with the escape-char:
to-single-quotes("c\\") returns "c". if you have an idea on how to resolve it, we can get rid of this ugly callback